### PR TITLE
Formula: add full_name variable

### PR DESCRIPTION
**Pull request by xu-cheng**
cc @Homebrew/owners @bfontaine @DomT4

### CODE REVIEW

**Comment by DomT4** [comment]
Looks good here with some testing. Didn't hit anything unexpected.

File: Library/Homebrew/cmd/uses.rb [hunk 1]
```diff
@@ -25,7 +25,7 @@ def uses
             reqs = f.recursive_requirements do |dependent, req|
               Requirement.prune if ignores.any? { |ignore| req.send(ignore) } && !dependent.build.with?(req)
             end
-            deps.any? { |dep| dep.to_formula.name == ff.name } ||
+            deps.any? { |dep| dep.to_formula.full_name == ff.full_name } ||
               reqs.any? { |req| req.name == ff.name || req.class.default_formula == ff.name }
```
Comment (xu-cheng) [inline Library/Homebrew/cmd/uses.rb]: I'm not sure whether it should be name or full_name here. Or maybe just:
```ruby
reqs.any? { |req| req.name == ff.name || req.class.default_formula == ff.name || req.class.default_formula == ff.full_name }
```

**Comment by xu-cheng** [comment]
Any other comments?

**Comment by mikemcquaid** [comment]
Looks good :+1:

**Comment by jacknagel** [comment]
lgtm

**Pull request merged**